notification/method/webpush.php

```diff
@@ -428,4 +433,81 @@ protected function clean_expired_subscriptions(array $user_subscription_map, arr
 
 		$this->remove_subscriptions($remove_subscriptions);
 	}
+
+	/**
+	 * Takes an avatar string (usually they are in full html format already) and
+	 * extracts the image src tag url code. Then, if the avatar string is a relative
+	 * path, it's converted to absolute path.
+	 *
+	 * Converts:
+	 * 		<img class="avatar" src="./path/to/avatar=123456789.gif" width="123" height="123" alt="User avatar" />
+	 * 	or  <img class="avatar" src="./styles/prosilver/theme/images/no_avatar.gif" data-src="./path/to/avatar=123456789.gif" width="123" height="123" alt="User avatar" />
+	 * into https://myboard.url/path/to/avatar=123456789.gif
+	 *
+	 * @param string $avatar
+	 * @return string Absolute path to avatar image
+	 */
+	protected function prepare_avatar($avatar)
+	{
+		$pattern = '/src=["\']?([^"\'>]+)["\']?/';
+
+		if (preg_match($pattern, $avatar, $matches))
+		{
+			$avatar = $matches[2] ?? $matches[1];
+		}
+
+		if (strpos($avatar, 'download/file'))
+		{
+			$avatar = $this->get_uploaded_avatar($avatar);
+		}
+
+		return preg_replace("#^\\{$this->path_helper->get_web_root_path()}#", $this->get_board_url(), $avatar, 1);
+	}
+
+	/**
+	 * Get the actual path to uploaded avatars
+	 *
+	 * @param string $file
+	 * @return string
+	 */
+	protected function get_uploaded_avatar($file)
+	{
+		$prefix = $this->config['avatar_salt'] . '_';
+		$image_dir = rtrim($this->config['avatar_path'], DIRECTORY_SEPARATOR);
+
+		// Ensure image_dir is relative
+		$image_dir = str_replace(['../', '..\\', './', '.\\'], '', $image_dir);
+		if ($image_dir && ($image_dir[0] === '/' || $image_dir[0] === '\\'))
+		{
+			$image_dir = '';
+		}
+
+		// Extract file extension
+		$ext = pathinfo($file, PATHINFO_EXTENSION);
+
+		// Extract filename using regex
+		$filename = '';
+		if (preg_match('/avatar=(.+?)_/', $file, $matches))
+		{
+			$filename = $matches[1];
+		}
+
+		// Construct final path
+		return $filename ? $this->path_helper->get_web_root_path() . $image_dir . DIRECTORY_SEPARATOR . $prefix . $filename . '.' . $ext : '';
```

Comment:
`DIRECTORY_SEPARATOR` is missing between `$this->path_helper->get_web_root_path()` and `$image_dir`. Also tested it on Windows and resulting path contains slashes and backslashes messed up, like
`http://localhost/33\images/avatars/upload\903172b946f8ce7e670825aa24ee8ed4_2.jpg`.